### ISSUE TITLE
How to pass in step argument to `scheduler.step()`

### ISSUE THREAD

**Issue by benihime91** [PyTorchLightning/pytorch-lightning#5708]
How does one pass in `step` argument to `scheduler.step()`. 
What i want to achieve is something like this -
```python
for epoch in range(...):
    for batch_nb in ....
         ....
        scheduler.step((epoch + 1)* batch_nb)
```
or 
```python
for epoch in range(...) :
    ....
   scheduler.step(epoch+1)
```
The one thing that i found was for `ReduceLROnPlateau` we can pass in `monitor` in the `scheduler_step`. So if i do this :
```python
def training_step(**kwargs):
      ...
     self.log("my_step", self.global_step, on_step=True, on_epoch=False, progress_bar=False, logger=False)
     self.log("my_epoch", self.current_epoch, on_step=False, on_epoch=False, progress_bar=False, logger=False)
```
within my `LightningModule` & then construct my scheduler (s) like so -
```python
scheduler = dict(scheduler=scheduler, frequency=1, interval="step", monitor="my_step", strict=True)
scheduler = dict(scheduler=scheduler, frequency=1, interval="epoch", monitor="my_epoch",  strict=True)
```
Will this work ?
